tools/cldr-code/src/main/java/org/unicode/cldr/tool/Chart.java

```diff
@@ -165,7 +184,19 @@ public static String getTsvDir(String targetDir, String topicName) {
     }
 
     public String getFixLinkFromPath(CLDRFile cldrFile, String path) {
-        String result = PathHeader.getLinkedView(surveyUrl, cldrFile, path);
+        String result = PathHeader.getLinkedView(CLDRConfig.getInstance().urls(), cldrFile, path);
         return result == null ? "" : result;
     }
+
+    /**
+     * Attempt to allocate the Chart that goes along with this report
+     */
+    public static Chart forReport(final ReportId report, final String locale) {
+        switch(report) {
+            case personnames:
+                return new ChartPersonName(locale);
+            default:
+                return null;
+        }
+    }
```

Comment:
@macchiati  maps a `ReportId` enum to a Chart object.  That should be all that is needed to add a new report, once the JS side is updated.

Reply:
Very cool!